srk/src/polynomialConeCpClosure.ml

```diff
@@ -188,79 +196,132 @@ let compute_transformation affine_basis ctxt : transformation_data =
     (pp_transformation_data pp_dim) data;
   data
 
+let polyhedron_of ctxt zeroes positives =
+  L.logf ~level:`trace
+    "polyhedron_of: conversion context for polyhedron is: @[%a@]@;"
+    (PolyVectorContext.pp pp_dim)
+    ctxt;
+  let to_vector = PolyVectorConversion.poly_to_vector ctxt in
+  let (linear_constraints, conic_constraints) =
+    ( List.map (fun poly -> (`Zero, to_vector poly)) zeroes
+    , List.map (fun poly -> (`Nonneg, to_vector poly)) positives ) in
+  let p = Polyhedron.of_constraints
+            (BatList.enum (List.append linear_constraints conic_constraints)) in
+  L.logf ~level:`trace
+    "@[polyhedron_of: @[<v 0>zeroes: @[<v 0>%a@]@; positives: @[%a@]@]@; is: %a@]@;"
+    (pp_poly_list pp_dim) zeroes
+    (pp_poly_list pp_dim) positives
+    (Polyhedron.pp pp_dim) p;
+  p
+
+(**
+    Given a regular polynomial cone C and a subspace U spanned by
+    [variables] and 1, such that the monomial ordering of C is a
+    graded elimination order that retains [variables] (e.g., devreglex),
+    compute the polyhedron [C \cap U].
+ *)
+let intersect_cone_and_affine_space cone variables =
```

Comment:
This function is "PolynomialCone.restrict (fun m -> Monomial.degree m <= 1) cone", except that for non-proper cones you get the zeros [1; x; y; z] instead of just [1].  Do you need that behavior?